File: problem-solving/baekjoon/step18/11047_동전_0.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n, k int
	fmt.Fscanln(reader, &n, &k)
	var values = make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscanln(reader, &values[i])
	}
	minCount := greedyAlgorithm(values, k, n)
	fmt.Println(minCount)
}

func greedyAlgorithm(values []int, k, n int) (coinCount int) {
	for i := n - 1; i >= 0; i-- {
		coinCount = coinCount + k/values[i]
		k %= values[i]
	}
	return
}

File: problem-solving/baekjoon/11047_동전_0.py
import sys

def greedy_algorithm(values, k):
    coin_count = 0
    for i in reversed(range(0, n)):
        coin_count = coin_count + int(k / values[i])
        k %= values[i]
    return coin_count

if __name__ == "__main__":
    n, k = map(int, sys.stdin.readline().split()) 
    values = []
    for i in range(0, n):
        values.append(int(sys.stdin.readline()))
    
    min_count = greedy_algorithm(values, k)
    print(min_count)